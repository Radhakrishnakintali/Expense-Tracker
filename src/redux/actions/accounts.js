export function deleteTransaction(id){
    return (dispatch) => {
        dispatch({
            type: "DELETE_TRANSACTION",
            payload: id
        });
    }
}

export function addTransaction(transaction){
    return (dispatch) => {
        dispatch({
            type: "ADD_TRANSACTION",
            payload: transaction
        });
    }
}

export function updateTransaction(updatedTransaction) {
    return (dispatch) => {
        dispatch({
            type: "UPDATE_TRANSACTION",
            payload: updatedTransaction
        });
    }
}