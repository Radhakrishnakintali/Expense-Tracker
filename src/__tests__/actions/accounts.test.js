import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import * as selectActions from '../../redux/actions/accounts';

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);
const store = mockStore();

beforeEach(() => { // Runs before each test in the suite
    store.clearActions();
});

test('Dispatches the correct deleteTransaction action and payload', () => {
    const expectedAction = [
      {
        'payload': 1,
        'type': 'DELETE_TRANSACTION',
      },
    ];

    store.dispatch(selectActions.deleteTransaction(1));
    expect(store.getActions()).toEqual(expectedAction);
});

test('Dispatches the correct action and payload1', () => {
    store.dispatch(selectActions.deleteTransaction(1));
    expect(store.getActions()).toMatchSnapshot();
});

test('Dispatches the correct action and payload', () => {
    const expectedAction = [
      {
        'payload': {id:1,text:"data",amount:200},
        'type': 'ADD_TRANSACTION',
      },
    ];

    store.dispatch(selectActions.addTransaction({id:1,text:"data",amount:200}));
    expect(store.getActions()).toEqual(expectedAction);
});

test('Dispatches the correct update transaction action and payload', () => {
    const expectedAction = [
      {
        'payload': {id:1,text:"data",amount:200},
        'type': 'UPDATE_TRANSACTION',
      },
    ];

    store.dispatch(selectActions.updateTransaction({id:1,text:"data",amount:200}));
    expect(store.getActions()).toEqual(expectedAction);
});