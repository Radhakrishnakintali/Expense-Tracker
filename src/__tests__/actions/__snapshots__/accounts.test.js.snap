// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dispatches the correct action and payload1 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "DELETE_TRANSACTION",
  },
]
`;
